packages/cli/src/commands/account/authorize-l2.test.ts

```diff
@@ -0,0 +1,239 @@
+import { testWithAnvilL2 } from '@celo/dev-utils/lib/anvil-test'
+import { addressToPublicKey } from '@celo/utils/lib/signatureUtils'
+import Web3 from 'web3'
+import { testLocallyWithWeb3Node } from '../../test-utils/cliUtils'
+import { PROOF_OF_POSSESSION_SIGNATURE } from '../../test-utils/constants'
+import Lock from '../lockedgold/lock'
+import ValidatorRegister from '../validator/register'
+import Authorize from './authorize'
+import Register from './register'
+
+process.env.NO_SYNCCHECK = 'true'
+
+testWithAnvilL2('account:authorize cmd', (web3: Web3) => {
+  test('can authorize vote signer', async () => {
+    const accounts = await web3.eth.getAccounts()
+    const notRegisteredAccount = accounts[0]
+    const signerNotRegisteredAccount = accounts[1]
+
+    await testLocallyWithWeb3Node(Register, ['--from', notRegisteredAccount], web3)
+
+    await testLocallyWithWeb3Node(
+      Authorize,
+      [
+        '--from',
+        notRegisteredAccount,
+        '--role',
+        'vote',
+        '--signer',
+        signerNotRegisteredAccount,
+        '--signature',
+        PROOF_OF_POSSESSION_SIGNATURE,
+      ],
+      web3
+    )
+  })
+
+  test('can authorize attestation signer', async () => {
+    const accounts = await web3.eth.getAccounts()
+    const notRegisteredAccount = accounts[0]
+    const signerNotRegisteredAccount = accounts[1]
+
+    await testLocallyWithWeb3Node(Register, ['--from', notRegisteredAccount], web3)
+    await testLocallyWithWeb3Node(
+      Authorize,
+      [
+        '--from',
+        notRegisteredAccount,
+        '--role',
+        'attestation',
+        '--signer',
+        signerNotRegisteredAccount,
+        '--signature',
+        PROOF_OF_POSSESSION_SIGNATURE,
+      ],
+      web3
+    )
+  })
+
+  test('can authorize validator signer before validator is registered', async () => {
+    const accounts = await web3.eth.getAccounts()
+    const notRegisteredAccount = accounts[0]
+    const signerNotRegisteredAccount = accounts[1]
+
+    await testLocallyWithWeb3Node(Register, ['--from', notRegisteredAccount], web3)
+    await testLocallyWithWeb3Node(
+      Authorize,
+      [
+        '--from',
+        notRegisteredAccount,
+        '--role',
+        'validator',
+        '--signer',
+        signerNotRegisteredAccount,
+        '--signature',
+        PROOF_OF_POSSESSION_SIGNATURE,
+      ],
+      web3
+    )
+  })
+
+  // TODO figure out how we tackle this failure
```

Comment:
👀

Reply:
this will have to be revisited while working on https://github.com/celo-org/developer-tooling/issues/417